presto-native-execution/presto_cpp/main/TaskResource.cpp

```diff
@@ -221,6 +250,47 @@ proxygen::RequestHandler* TaskResource::createOrUpdateTaskImpl(
       });
 }
 
+proxygen::RequestHandler* TaskResource::validateBatchQueryPlan(
+    proxygen::HTTPMessage* message,
+    const std::vector<std::string>& pathMatch) {
+  return new http::CallbackRequestHandler(
+      [this](
+          proxygen::HTTPMessage* /*message*/,
+          const std::vector<std::unique_ptr<folly::IOBuf>>& body,
+          proxygen::ResponseHandler* downstream) {
+        std::ostringstream oss;
+        for (auto& buf : body) {
+          oss << std::string((const char*)buf->data(), buf->length());
+        }
+        std::string request = oss.str();
+        protocol::BatchPlanValidationRequest planValidationRequest =
+            json::parse(request);
+        const protocol::TaskId taskId{"0"};
```

Comment:
This may not work. See AssignUniqueId plan node. Can we use properly formatted task ID?